strr-examiner-web/app/composables/useExaminerActions.ts

```diff
@@ -22,15 +23,27 @@ export const useExaminerActions = () => {
     buttonPosition: 'left' | 'right',
     buttonIndex: number,
     refresh: () => void,
-    additionalArgs: Args = [] as unknown as Args
+    additionalArgs: Args = [] as unknown as Args,
+    validateFn?: () => Promise<boolean> | MultiFormValidationResult | boolean
   ) => {
     try {
       handleButtonLoading(false, buttonPosition, buttonIndex)
+      if (validateFn) {
+        let isValid = true
+        if (action === 'SEND_NOC') {
+          isValid = await validateFn() as boolean
+        }
+        if (!isValid) {
+          handleButtonLoading(true)
```

Comment:
The `finally` block has the same call and would take care of the button's state, so this is redundant.

Reply:
Done